modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/outputs/EnrichedEvent.scala

```diff
@@ -246,9 +247,163 @@ class EnrichedEvent extends Serializable {
 
   // Fields modified in PII enrichemnt (JSON String)
   @BeanProperty var pii: String = _
+
+  private val JsonSchemaDateTimeFormat =
+    DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS")
+
+  def toKv(k: String, v: String): Seq[(String, Json)] = if (v != null) Seq((k, Json.fromString(v))) else Seq.empty[(String, Json)]
+  def toKv(k: String, v: JByte): Seq[(String, Json)] = if (v != null) Seq((k, Json.fromBoolean(v == 1))) else Seq.empty[(String, Json)]
+  def toKv(k: String, v: JInteger): Seq[(String, Json)] = if (v != null) Seq((k, Json.fromInt(v))) else Seq.empty[(String, Json)]
+  def toKv(k: String, v: JFloat): Seq[(String, Json)] = if (v != null) Seq((k, Json.fromFloatOrNull(v))) else Seq.empty[(String, Json)]
```

Comment:
You can write a generic version.

```suggestion
  def toKv[T](k: S, v: T, f: T => Json): Option[(String, Json)] =
    Option(v).map(value => (k, f(value)))
```
And then reuse it, e.g:
```
def toKv(k :String, b: JInteger): Option[(String, Json)]  = toKv(k, b, Json.fromInt)
```
You can call directly `.toSeq` on an `Option` so you can have e.g:
```
  toKv("user_id", this.user_id).toSeq
```
or put the `toSeq` directly in the above function

Reply:
I considered this, but there are issues with java/scala types. Working around them is more code, than current implementaion.
```
found   : Int => io.circe.Json
 required: Integer => io.circe.Json
  def toKv(k: String, v: JInteger): Seq[(String, Json)] = toKv[JInteger](k, v, Json.fromInt)

  found   : scala.Float => io.circe.Json
 required: java.lang.Float => io.circe.Json
  def toKv(k: String, v: JFloat): Seq[(String, Json)] = toKv[JFloat](k, v, Json.fromFloatOrNull)
  ```

Reply:
FYI I updated it

https://github.com/snowplow/enrich/pull/521/files#diff-a546d8f6e12feda9bb60528ceb0fc7160b3d388f429e0a2299e488fd044128aaR258-R274